pkgs/os-specific/linux/iproute/default.nix

```diff
@@ -26,8 +26,9 @@ stdenv.mkDerivation rec {
     "HDRDIR=$(TMPDIR)/include/iproute2" # Don't install headers
   ];
 
+  # enable iproute2 module if you want this folder to be created
   buildFlags = [
-    "CONFDIR=/etc/iproute2"
+    "CONFDIR=${config.networking.iproute2.confDir or "/run/iproute2"}"
```

Comment:
Can you verify if this works? That is can you change confDir and get a different value passed to buildFlags? I'm not 100% on this but I think you need to directly pass it with ```nixpkgs.config``` in your module (see https://github.com/NixOS/nixpkgs/blob/release-18.03/nixos/modules/misc/nixpkgs.nix#L92-L106). Nixpkgs & NixOS have different config sets which can be confusing.

Reply:
IMHO this requires a change log entry.

I am also not so sure if changing the default to `/run` is a good decision. If someone needs this the configuration will be set to whatever the user wants and the default behavior should not be altered if not used.

Am I missing something?

Reply:
I agree this can be cool to have as a changelogentry. How would I do that ? other PR ? what document ?
I don't see a usecase either to let the user set the path (he would lose the cache) so I am tempted to remove the `confDir = mkOption {` but the paths between the module and the package must be in sync so I would let the nixpkgs.config.iproute2.confDir. If the user can't choose confDir, then we need to agree on the default. /etc/iproute2 is the common one but I guess /run/iproute2 matches dynamic files better. I am fine either way.

Reply:
Ok yeah just using /run/iproute2 makes sense.

Reply:
It's fine now

Reply:
Why not to keep "/etc/iproute2" as the default in the case if ```config.networking.iproute2.confDir``` ?

I have a nixos module which creates files in "/etc/iproute2"

Reply:
> /etc/iproute2 is the common one but I guess /run/iproute2 matches dynamic files better. I am fine either way.

Besides the compatibility with older Nixos and with the scripts made for other Linuxes, "/etc/iproute2" has the advantage of being configured via
```
environment.etc."iproute2".text = ''blablabla'';
```
while modification of "/run/iproute2" would require a ```systemd``` module with a carefully set up start order.